src/argilla/client/feedback/dataset/local/dataset.py

```diff
@@ -282,3 +286,197 @@ def delete_metadata_properties(
+            ),
+        ):
+            raise ValueError(f"Training data {type(task)} is not supported yet")
+
+        data = task._format_data(local_dataset)
+        if framework in [
+            Framework.TRANSFORMERS,
+            Framework.SETFIT,
+            Framework.SPAN_MARKER,
+            Framework.PEFT,
+        ]:
+            return task._prepare_for_training_with_transformers(
+                data=data, train_size=train_size, seed=seed, framework=framework
+            )
+        elif framework in [Framework.SPACY, Framework.SPACY_TRANSFORMERS]:
+            require_dependencies("spacy")
+            import spacy
+
+            if lang is None:
+                _LOGGER.warning("spaCy `lang` is not provided. Using `en`(English) as default language.")
+                lang = spacy.blank("en")
+            elif lang.isinstance(str):
+                if len(lang) == 2:
+                    lang = spacy.blank(lang)
+                else:
+                    lang = spacy.load(lang)
+            return task._prepare_for_training_with_spacy(data=data, train_size=train_size, seed=seed, lang=lang)
+        elif framework is Framework.SPARK_NLP:
+            return task._prepare_for_training_with_spark_nlp(data=data, train_size=train_size, seed=seed)
+        elif framework is Framework.OPENAI:
+            return task._prepare_for_training_with_openai(data=data, train_size=train_size, seed=seed)
+        elif framework is Framework.TRL:
+            return task._prepare_for_training_with_trl(data=data, train_size=train_size, seed=seed)
+        elif framework is Framework.TRLX:
+            return task._prepare_for_training_with_trlx(data=data, train_size=train_size, seed=seed)
+        elif framework is Framework.SENTENCE_TRANSFORMERS:
+            return task._prepare_for_training_with_sentence_transformers(data=data, train_size=train_size, seed=seed)
+        else:
+            raise NotImplementedError(
+                f"Framework {framework} is not supported. Choose from: {[e.value for e in Framework]}"
+            )
+
+    def update_records(self, records: Union["FeedbackRecord", List["FeedbackRecord"]]) -> None:
+        warnings.warn(
+            "`update_records` method only works for `FeedbackDataset` pushed to Argilla. "
+            "If your are working with local data, you can just iterate over the records and update them."
+        )
+
+    def sort_by(
+        self, field: Union[str, RecordSortField], order: Union[str, SortOrder] = SortOrder.asc
+    ) -> "FeedbackDataset":
+        warnings.warn(
+            "`sort_by` method is not supported for local datasets and won't take any effect. "
+            "First, you need to push the dataset to Argilla with `FeedbackDataset.push_to_argilla()`. "
+            "After, use `FeedbackDataset.from_argilla(...).sort_by()`.",
+            UserWarning,
+            stacklevel=1,
+        )
+        return self
+
+    def pull(self) -> "FeedbackDataset":
```

Comment:
Dito (plus a `DeprecationWarning` on `pull`)
```suggestion
    def pull_from_argilla(self) -> "FeedbackDataset":
```